### PR TITLE
[studio-ui] Add highlight to Sidebar #2901

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/2901

### CODE REVIEW

File: static-assets/components/cstudio-common/common-api.js [hunk 1]
```diff
@@ -385,6 +385,10 @@ var nodeOpen = false,
 
                 var event = new CustomEvent("setContentDone");
                 document.dispatchEvent(event);
+
+                amplify.publish("SET_TREE_HIGHLIGHT", {
```
Comment (rart) [inline static-assets/components/cstudio-common/common-api.js]: The event should rather be `SELECTED_CONTENT_SET`

File: static-assets/components/cstudio-common/common-api.js [hunk 2]
```diff
@@ -3010,6 +3014,46 @@ var nodeOpen = false,
...
                 return path;
+            },
+
+            /**
+             * updates sidebar tree cookie that keeps its state
+             * @param {*} tree - tree that the cookie will be updated
+             * @param {*} cookieKey - cookey key of the tree that will be updated (depends of tree)
+             * @param {*} path - path to be added into the tree cookie (state) -> item uri
+             */
+            updateTreePath: function(treeName, cookieKey, path){
```
Comment (rart) [inline static-assets/components/cstudio-common/common-api.js]: rename to `updateTreeCookiePath`

File: static-assets/components/cstudio-contextual-nav/wcm-site-dropdown-mods/wcm-assets-folder.js [hunk 3]
```diff
@@ -487,7 +487,16 @@ CStudioAuthoring.ContextualNav.WcmAssetsFolder = CStudioAuthoring.ContextualNav.
      * render a tree item
      */
     drawTreeItem: function(treeNodeTO, root, instance) {
-        var WcmAssets = CStudioAuthoring.ContextualNav.WcmAssetsFolder;
+        var WcmAssets = CStudioAuthoring.ContextualNav.WcmAssetsFolder,
```
Comment (rart) [inline static-assets/components/cstudio-contextual-nav/wcm-site-dropdown-mods/wcm-assets-folder.js:490]: Was this file meant to be modified? You were originally working with `wcm-root-folder.js`.

File: static-assets/components/cstudio-contextual-nav/wcm-site-dropdown-mods/wcm-root-folder.js [hunk 4]
```diff
@@ -890,17 +900,32 @@
...
                     treeNode.nowrap = true;
 
+                    if(highlight){
+                        function loadedNode() {
+                            window.setTimeout(function(){
+                                self.scrollToHighlighted();
+                            }, 500);
+                        };
+
+                        treeNode.html.onload = loadedNode();
```
Comment (rart) [inline static-assets/components/cstudio-contextual-nav/wcm-site-dropdown-mods/wcm-root-folder.js]: Is this really what you wanted to do?

You're essentially setting `.onload = undefined` (setting the `.onload` to the return value of the `loadedNode` function which will always be undefined.

File: static-assets/components/cstudio-contextual-nav/wcm-site-dropdown-mods/wcm-root-folder.js [hunk 5]
```diff
@@ -1291,19 +1329,19 @@
 	                			node.isLeaf = true;
 	                		}
 
-	                		Self.drawSubtree(treeData.item.children, args.node, args.pathToOpenTo, args.instance);
```
Comment (rart) [inline static-assets/components/cstudio-contextual-nav/wcm-site-dropdown-mods/wcm-root-folder.js:1294]: @sumerjabri this is another reason to call up GitHub support. This part wasn't really changed in content but it's getting modified marks and as a reviewer is confusing.

File: static-assets/components/cstudio-contextual-nav/wcm-site-dropdown-mods/wcm-root-folder.js [hunk 6]
```diff
@@ -3485,6 +3521,25 @@
         this.onClickAction = (config.params["onClick"]) ? config.params["onClick"] : "";
         this.config = config;
         this.mods = [];
+       
+        if(config.params.path === '/site/website'){            
+            amplify.subscribe("SET_TREE_HIGHLIGHT", function (message) {
```
Comment (rart) [inline static-assets/components/cstudio-contextual-nav/wcm-site-dropdown-mods/wcm-root-folder.js]: Change event name as suggested on common-api comment.

File: static-assets/scripts/crafter.js [hunk 7]
```diff
@@ -77,6 +77,7 @@
         "GUEST_SITE_URL_CHANGE": "GUEST_SITE_URL_CHANGE",
         "CHANGE_GUEST_REQUEST": "CHANGE_GUEST_REQUEST",
         "IS_REVIEWER": "IS_REVIEWER",
+        "SET_TREE_HIGHLIGHT": "SET_TREE_HIGHLIGHT",
```
Comment (rart) [inline static-assets/scripts/crafter.js]: Change name as suggested

**Review by rart — CHANGES_REQUESTED** [review]

File: static-assets/components/cstudio-contextual-nav/wcm-site-dropdown-mods/wcm-assets-folder.js (hunk 3 above)
Comment (jvega190) [reply static-assets/components/cstudio-contextual-nav/wcm-site-dropdown-mods/wcm-assets-folder.js:490]: We need the highlight in both root and assets folders, for example, assets like images can be previewed in studio, while being previewed they should be highlighted

**Review by rart — APPROVED** [review]

**Pull request merged**